[LintDiff] Fix correlation: tags are strings, not Date

danieljurek opened the pull request:
Configure YAML parser to return basic JSON-compatible objects. 

Legacy LintDiff: https://github.com/Azure/azure-rest-api-specs/pull/34379/checks?check_run_id=41443910398
New LintDiff: https://github.com/Azure/azure-rest-api-specs/actions/runs/14761776132

danieljurek committed:
Fix correlation: YAML parsing for tag will sometimes return a Date object. Configure the YAML parser to return a string instead.

mikeharder commented on eng/tools/lint-diff/src/markdown-utils.ts:
```diff
@@ -200,7 +200,7 @@ export function getDefaultTag(markdownContent: string): string {
   const lh = codeBlockMap[latestHeader];
   if (lh) {
-    const latestDefinition = YAML.load(lh.literal!) as undefined | { tag: string };
+    const latestDefinition = YAML.load(lh.literal!, { schema: YAML.JSON_SCHEMA }) as undefined | { tag: string };
```
```suggestion
    const latestDefinition = YAML.load(lh.literal!, { schema: YAML.JSON_SCHEMA }) as undefined | { tag: string };
```
```suggestion
    const latestDefinition = YAML.load(lh.literal!, { schema: YAML.FAILSAFE_SCHEMA }) as undefined | { tag: string };
```

Let's go all the way and just use the FAILSAFE_SCHEMA unless we know we need more.

https://teams.microsoft.com/l/message/19:987876ab02634aaeaded0fb0de9f338a@thread.skype/1744173886127?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1744162492642&teamName=Azure%20SDK&channelName=AutoRest%20-%20Engineering&createdTime=1744173886127

danieljurek replied:
From the conversation...

> I don't think we want anything getting interpreted as a number or boolean 

Makes sense!

mikeharder commented on eng/tools/lint-diff/src/markdown-utils.ts:
```diff
@@ -210,7 +210,9 @@ export function getDefaultTag(markdownContent: string): string {
         continue;
       }
-      const latestDefinition = YAML.load(lh.literal!) as undefined | { tag: string };
+      const latestDefinition = YAML.load(lh.literal!, { schema: YAML.JSON_SCHEMA }) as
```
```suggestion
      const latestDefinition = YAML.load(lh.literal!, { schema: YAML.FAILSAFE_SCHEMA }) as
```

mikeharder reviewed (changes requested):
see comments

danieljurek committed:
Apply suggestions from code review

Use FAILSAFE_SCHEMA which supports strings, arrays, and plain objects which prevents parsing and interpretation of various number formats, booleans, null, etc.

Co-authored-by: Mike Harder <mharder@microsoft.com>

danieljurek commented:
This can merge. Bypassing protections.
